core/src/main/java/org/opensearch/sql/analysis/HighlightAnalyzer.java

```diff
@@ -34,8 +34,10 @@ public LogicalPlan visitAlias(Alias node, AnalysisContext context) {
       return null;
     }
 
+    // Must set name here else operator cannot resolve delegated reference
+    ((HighlightFunction) node.getDelegated()).setName(node.getName());
```

Comment:
can we avoid calling node.getDelegated() 3 times? Save it to a variable and cast it only once.